### PR TITLE
obliterates some unneeded vars

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
obliterates some unneeded vars, guess who needs to do list debugging soon

## Why It's Good For The Game
memory optimization

## Testing
Compiled, ran, most vars worked fine

## Changelog
NPFC

### CODE REVIEW

File: code/datums/datumvars.dm [hunk 1]
```diff
@@ -32,21 +32,9 @@
 	switch(var_name)
 		if("vars")
 			return FALSE
-		if("var_edited")
+		if(TRAIT_VAR_EDITED)
```
Comment (Contrabang) [inline code/datums/datumvars.dm:35]: A trait isnt a var, this isnt going to do anything. If anything I think we should do what TG did and turn these into flags

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/datums/datumvars.dm (hunk 1 above)
Comment (GDNgit) [reply code/datums/datumvars.dm:35]: I know, I need to do it but I can just put it on the list edits

**Comment by ATP-Engineer** [comment]
image

**Comment by datlo** [comment]
Another unused variable is the 'school' variable for spells. 

If we ever want to do anything related to spell schools we are going to have to reclassify all of them anyway since it's never been properly maintained/looked at, so it'd be good riddance to remove them entirely.

File: code/__HELPERS/trait_helpers.dm [hunk 2]
```diff
@@ -134,6 +134,12 @@
 Remember to update _globalvars/traits.dm if you're adding/removing/renaming traits.
 */
 
+//Datum Traits
+/// this datum is being processed in SSobj
```
Comment (SteelSlayer) [inline code/__HELPERS/trait_helpers.dm:138]: Can be for other SS too, like SSfastprocess
```suggestion
/// this datum is being processed in a subsystem
```

File: code/__HELPERS/path.dm [hunk 3]
```diff
@@ -395,16 +395,8 @@
...
-	/// These are generally cheaper than looping contents so they go first
-	switch(destination_turf.pathing_pass_method)
-		if(TURF_PATHING_PASS_DENSITY)
-			if(destination_turf.density)
-				return TRUE
-		if(TURF_PATHING_PASS_PROC)
-			if(!destination_turf.CanPathfindPass(ID, actual_dir, caller, no_id = no_id))
-				return TRUE
-		if(TURF_PATHING_PASS_NO)
-			return TRUE
```
Comment (SteelSlayer) [inline code/__HELPERS/path.dm:407]: Comment seems to suggest this can make things cheaper. Why remove these? Do they not work?

Comment (GDNgit) [reply code/__HELPERS/path.dm:407]: It's never set to a different value so it always defaults to the density check

File: code/controllers/subsystem/SStimer.dm [hunk 4]
```diff
@@ -418,8 +418,8 @@ SUBSYSTEM_DEF(timer)
 	if ((timeToRun < world.time || timeToRun < SStimer.head_offset) && !(flags & TIMER_CLIENT_TIME))
 		CRASH("Invalid timer state: Timer created that would require a backtrack to run (addtimer would never let this happen): [SStimer.get_timer_debug_string(src)]")
 
-	if (callBack.object != GLOBAL_PROC && !QDESTROYING(callBack.object))
-		LAZYADD(callBack.object.active_timers, src)
+	if(callBack.object != GLOBAL_PROC && flags & TIMER_DELETE_ME && !QDESTROYING(callBack.object))
```
Comment (Contrabang) [inline code/controllers/subsystem/SStimer.dm:421]:
```suggestion
	if(callBack.object != GLOBAL_PROC && (flags & TIMER_DELETE_ME) && !QDESTROYING(callBack.object))
```

File: code/datums/datumvars.dm [hunk 5]
```diff
@@ -32,21 +32,9 @@
...
-
-
-/client/vv_edit_var(var_name, var_value) //called whenever a var is edited
-	switch(var_name)
-		if("vars")
-			return FALSE
-		if("var_edited")
-			return FALSE
-	var_edited = TRUE
+	ADD_TRAIT(src, TRAIT_VAR_EDITED, TRAIT_SOURCE_ADMIN)
```
Comment (Contrabang) [inline code/datums/datumvars.dm:37]: I think for these low-level things for each datum, we should do what TG did and apply them to a datum_flags var. That is the best way to protect them from stuff like this, but also allows for cutting down on the amount of datum vars

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/datums/datumvars.dm (hunk 5 above)
Comment (GDNgit) [reply code/datums/datumvars.dm:37]: potentially? But it runs into the weird situation of "What if we want some stuff edited and we don't want others edited".

Comment (Contrabang) [reply code/datums/datumvars.dm:37]: I think all cases of stuff that TG's added to their datum vars would be safe to add to ours, i.e. var edited, is processing, etc. I dont think we ever want people messing around with datum vars really anyways

**Review by S34NW — CHANGES_REQUESTED** [review]
Outstanding changes since nearly day one, it's time this got sorted or got out.

**Pull request closed without merging**